src/expression_pruner.cpp

```diff
@@ -0,0 +1,77 @@
+#include "expression_pruner.h"
+#include "ir/visitor.h"
+#include "pruner_util.h"
+
+namespace P4PRUNER {
+
+IR::Node *ExpressionPruner::preorder(IR::Operation_Binary *s) {
+
+    IR::Node *to_remove =
+        rand() < 0.5 ? (IR::Node *)s->left : (IR::Node *)s->right;
+    visit(to_remove);
+}
+
+IR::Node *ExpressionPruner::preorder(IR::Operation_Unary *s) {
```

Comment:
The same as discussed with Operation_binary applies to Operation_Unary. We need to decide on a case by case basis on how to handle the expression.